component/wallet-cli/pkg/walletrunner/wallet_runner_oidc4ci.go

```diff
@@ -229,11 +232,192 @@ func (s *Service) RunOIDC4CI(config *OIDC4CIConfig, hooks *Hooks) error {
 	return nil
 }
 
+var matchRegex = regexp.MustCompile(oidc4ci.WalletInitFlowClaimRegex)
+
+func extractIssuerUrlFromScopes(scopes []string) (string, error) {
+	for _, scope := range scopes {
+		if matchRegex.MatchString(scope) {
+			return scope, nil
+		}
+	}
+
+	return "", errors.New("issuer url not found in scopes")
+}
+
+func (s *Service) RunOIDC4CIWalletInitiated(config *OIDC4CIConfig, hooks *Hooks) error {
+	log.Println("Starting OIDC4VCI authorized code flow Wallet initiated")
+	ctx := context.Background()
+	// Check whether scope contains combined string Issuer URL||ClaimEndpoint||credentialTemplateID
+	issuerUrl, err := extractIssuerUrlFromScopes(config.Scope)
+	if err != nil {
+		return errors.New(
+			"undefined scopes supplied. " +
+				"Make sure one of the provided scope is an vcs issuer url format. ref " +
+				oidc4ci.WalletInitFlowClaimRegex)
+	}
+
+	oidcIssuerCredentialConfig, err := s.getIssuerCredentialsOIDCConfig(
+		issuerUrl,
+	)
+	if err != nil {
+		return fmt.Errorf("get issuer oidc issuer config: %w", err)
```

Comment:
(instead of leaving a comment for each one, I'll just mention: for readability, it's better to capitalize acronyms whenever possible)